include/deal.II/lac/affine_constraints.templates.h

```diff
@@ -226,19 +226,46 @@ namespace internal
 
     // helper function
     const auto sort_constraints = [&]() {
+      if (locally_relevant_constraints.empty())
+        return;
+
       std::sort(locally_relevant_constraints.begin(),
                 locally_relevant_constraints.end(),
                 [](const auto &a, const auto &b) {
                   return std::get<0>(a) < std::get<0>(b);
                 });
 
-      locally_relevant_constraints.erase(
-        std::unique(locally_relevant_constraints.begin(),
-                    locally_relevant_constraints.end(),
-                    [](const auto &a, const auto &b) {
-                      return std::get<0>(a) == std::get<0>(b);
-                    }),
-        locally_relevant_constraints.end());
+      auto read_ptr  = locally_relevant_constraints.begin();
+      auto write_ptr = locally_relevant_constraints.begin();
+      while (++read_ptr != locally_relevant_constraints.end())
+        {
+          if (std::get<0>(*read_ptr) != std::get<0>(*write_ptr))
+            {
+              if (++write_ptr != read_ptr)
+                *write_ptr = std::move(*read_ptr);
+            }
+          else
+            {
+              auto &      a = *write_ptr;
+              const auto &b = *read_ptr;
+              Assert(std::get<0>(a) == std::get<0>(b), ExcInternalError());
+              if (std::get<1>(a) != std::get<1>(b))
+                std::get<1>(a) += std::get<1>(b);
+
+              auto &      av                = std::get<2>(a);
+              const auto &bv                = std::get<2>(b);
+              bool        vectors_are_equal = av.size() == bv.size();
+              for (unsigned int i = 0; vectors_are_equal && i < av.size(); ++i)
+                if (av[i].first != bv[i].first)
+                  vectors_are_equal = false;
```

Comment:
If we are at a `dof` index that is already present, we instead decide to merge the two lists, but only in case the two lists are different. This is to support cases where only the combined view between all processes can represent the final constraints.